workflow/controller/operator.go

```diff
@@ -3206,14 +3185,45 @@ func (woc *wfOperationCtx) setExecWorkflow() error {
 		woc.execWf = &wfv1.Workflow{Spec: *woc.wf.Status.StoredWorkflowSpec.DeepCopy()}
 		woc.volumes = woc.execWf.Spec.DeepCopy().Volumes
 	} else if woc.controller.Config.WorkflowRestrictions.MustUseReference() {
-		return fmt.Errorf("workflows must use workflowTemplateRef to be executed when the controller is in reference mode")
+		err := fmt.Errorf("workflows must use workflowTemplateRef to be executed when the controller is in reference mode")
+		woc.log.WithError(err).Errorf("Unable to get Workflow Template Reference for workflow")
+		woc.markWorkflowError(ctx, err)
```

Comment:
Why isn’t return error enough? The caller should be marking the work flow as error

Reply:
Some err will be `Error` and some are `Failure`. That's why it is making here. I will move the log into upper level